relayer/naive-strategy.go

```diff
@@ -232,6 +238,36 @@ func (nrs *NaiveStrategy) UnrelayedAcknowledgements(src, dst *Chain) (*RelaySequ
 
 // HandleEvents defines how the relayer will handle block and transaction events as they are emitted
 func (nrs *NaiveStrategy) HandleEvents(src, dst *Chain, events map[string][]string) {
+	// check for misbehaviour
+	if hdrs, ok := events[fmt.Sprintf("%s.%s", updateCliTag, headerTag)]; ok {
+		for i, hdr := range hdrs {
+			clientIDs := events[fmt.Sprintf("%s.%s", updateCliTag, clientIDTag)]
+
+			if src.PathEnd.ClientID == clientIDs[i] && dst.PathEnd.ClientID == clientIDs[i] {
+				hdrBytes, err := hex.DecodeString(hdr)
+				if err == nil {
+					exportedHeader, err := clienttypes.UnmarshalHeader(src.Encoding.Marshaler, hdrBytes)
+					if err == nil {
+						gotHeader, ok := exportedHeader.(*tmclient.Header)
+						if ok {
+							trustedHeader, err := src.GetLatestLightHeader()
+							if err == nil && !reflect.DeepEqual(gotHeader, trustedHeader) {
+								misbehaviour := tmclient.NewMisbehaviour(clientIDs[i], gotHeader, trustedHeader)
```

Comment:
before we pass in the headers. At this point the emittedHeader is good to go, but the header we created is missing trusted header info, this is where it gets tricky. If the relayer misses some blocks, it is possible it misses some previous misbehaving header, meaning the trusted header used in the emittedHeader might be invalid. 

We should create another helper function which gets all of the consensus states for this client, iterates through them and checks to see if they are valid (you should probably deduplicate some of the logic here). Once you find the earliest instance of misbehaviour. You need to reconstruct the misbehaving header via tx events and use the trusted header from the consensus state which occurs directly before that

This is kinda complicated, so lets leave it off for now and open an issue for it when this is ready to be merged